Use std::optional instead of absl::optional, which Chromium prefers.

Chromium commit 24ebcea5cee01 (crbug.com/40242125).

diff --git a/src/ntt_parameters.h b/src/ntt_parameters.h
index dc56ab62dd666..beceed2667210 100644
--- a/src/ntt_parameters.h
+++ b/src/ntt_parameters.h
@@ -169,7 +169,7 @@ struct NttParameters {
   ~NttParameters() = default;
 
   int number_coeffs;
-  absl::optional<ModularInt> n_inv_ptr;
+  std::optional<ModularInt> n_inv_ptr;
   std::vector<ModularInt> psis_bitrev;
   std::vector<ModularInt> psis_inv_bitrev;
   std::vector<unsigned int> bitrevs;
